src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -490,7 +489,7 @@ private Task ProcessCompilationUnitCompleted(CompilationUnitCompletedEvent compl
                     tasks.Add(task); 
                 }
 
-                return Task.WhenAll(tasks.ToArrayAndFree());
+                await Task.WhenAll(tasks.ToArrayAndFree()).ConfigureAwait(false);
```

Comment:
I think this was a bug. There is a finally block and I assume the intent was to execute the 'finally' only after all tasks have completed. Without the await, though, it'll run before the 'return'.